eli5/ipython.py

```diff
@@ -268,14 +282,51 @@ def show_prediction(estimator, doc, **kwargs):
     """
     format_kwargs, explain_kwargs = _split_kwargs(kwargs)
     expl = explain_prediction(estimator, doc, **explain_kwargs)
-    html = format_as_html(expl, **format_kwargs)
-    return HTML(html)
+    if expl.image is not None and expl.heatmap is not None:
+        # dispatch to image display implementation
+        return show_prediction_image(expl, **format_kwargs)
+    else:
+        # use default implementation
+        # TODO: test this
+        # TODO: a better design / refactorings might be needed
+        _set_html_kwargs_defaults(format_kwargs)
+        html = format_as_html(expl, **format_kwargs)
+        return HTML(html)
+
+
+def show_prediction_image(expl, **format_kwargs):
+    """ 
+    Show the heatmap and image overlay in a matplotlib plot 
+    
+    Returns
+    -------
+    figure and axes : tuple[object]
+        Matplotlib objects for the overlay.
+    """
+    # TODO: arguments to this show_prediction_image function 
+    # that control if any labels should be set, i.e. title
+    overlay = format_as_image(expl, **format_kwargs)
```

Comment:
Another option would be to return a PIL image here, then Jupyter notebook would already be able to display it, right? That would allow to remove matplotlib dependency if we don't need it otherwise. Or is it better to use matplotlib here?

Reply:
I think it's better to simply return the PIL image, which you can indeed display with `IPython.display.display()`. If the user wants a fancier plot they can use matplotlib with the returned PIL image.

(matplotlib will still however be a dependency through `formatters/image.py`).

Reply:
https://github.com/TeamHG-Memex/eli5/pull/315/commits/c357cbd4fcb245f5260436af8b434c223fb1ffca